### DISCUSSION TITLE
How To List HexoJS Posts By Updated Date

### DISCUSSION THREAD

**Discussion by dimaslanjaka** [hexojs/hexo#4920] [Question / Answer] [answered]
## ENV
hexo 6.0.0
node 12.x
npm latest

## Behaviour
i have code to iterate posts
```ejs
<% page.posts.each(function(post, i) { %>
  <% var year = post.date.year(); %>
  <!-- any element, example below with variable usage -->
  This post year is <%= year %> and last updated at <%= post.updated.toString(); %>
<% }); %>
```

Questions: how to order the `page.posts` by updated date descending ?
- iam reading about sorting an array [Stackoverflow](https://stackoverflow.com/questions/59421376/how-to-sort-array-based-on-date-in-descending-order-if-value-exists-in-js), but it causes **V8 Memory Leaking**.
- i've readed [https://hexo.io/docs/helpers#list-posts](https://hexo.io/docs/helpers#list-posts), but still dont know how to order by updated date because of less example todo it.
- i've tried `<% list_posts(page.posts, {'orderby': 'updated', 'order': -1}).each(function(post, i) { %>` but return error `list_posts(...).each is not a function` [github workflow log](https://github.com/dimaslanjaka/dimaslanjaka.github.io/runs/5701701807?check_suite_focus=true)

trying create helper function to order posts by updated date: 
```js
// how to order the posts in this function?
function list_post_by_updated(posts) {
  return posts;
}
/// usage <% list_post_by_updated(page.posts).each(function(post, i) { %>
hexo.extend.helper.register('list_post_by_updated', function (posts) {
  return list_post_by_updated(posts);
});
```

**Comment by stevenjoezhang** [comment, marked as answer]
In Hexo `_config.yml`, set
```
index_generator:
  path: ''
  per_page: 5
  order_by: -updated
```
the default value of `order_by` option is `-date`, change it to `-updated`

**Reply by dimaslanjaka** [reply]
thanks
